integrationpluginunipi.cpp

```diff
@@ -0,0 +1,1028 @@
+                ThingDescriptor.setParams(params);
+                info->addThingDescriptor(ThingDescriptor);
+            }
+        }
+
+        foreach (Neuron *neuron, m_neurons) {
+            ThingId ParentId = m_neurons.key(neuron);
+            foreach (QString circuit, neuron->digitalInputs()) {
+                ThingDescriptor ThingDescriptor(digitalInputThingClassId, QString("Digital input %1").arg(circuit), QString("Neuron %1").arg(neuron->type()), ParentId);
+                foreach(Thing *thing, myThings().filterByParentId(m_neurons.key(neuron))) {
+                    if (thing->paramValue(digitalInputThingCircuitParamTypeId) == circuit) {
+                        qCDebug(dcUniPi()) << "Found already added Circuit:" << circuit << ParentId;
+                        ThingDescriptor.setThingId(thing->id());
+                        break;
+                    }
+                }
+                ParamList params;
+                params.append(Param(digitalInputThingCircuitParamTypeId, circuit));
+                ThingDescriptor.setParams(params);
+                info->addThingDescriptor(ThingDescriptor);
+            }
+        }
+        return info->finish(Thing::ThingErrorNoError);
+    } else if (ThingClassId == digitalOutputThingClassId) {
+        foreach(Thing *Parent, myThings()) {
+            if ((Parent->thingClassId() == uniPi1ThingClassId) || (Parent->thingClassId() == uniPi1LiteThingClassId)) {
+                foreach (QString circuit, m_unipi->digitalOutputs()) {
+                    ThingDescriptor ThingDescriptor(digitalOutputThingClassId, QString("Digital output %1").arg(circuit), "UniPi 1", Parent->id());
+                    foreach(Thing *thing, myThings().filterByParentId(Parent->id())) {
+                        if (thing->paramValue(digitalOutputThingCircuitParamTypeId) == circuit) {
+                            qCDebug(dcUniPi()) << "Found already added Circuit:" << circuit << Parent->id();
+                            ThingDescriptor.setThingId(thing->id());
+                            break;
+                        }
+                    }
+                    ParamList params;
+                    params.append(Param(digitalOutputThingCircuitParamTypeId, circuit));
+                    ThingDescriptor.setParams(params);
+                    info->addThingDescriptor(ThingDescriptor);
+                }
+                break;
+            }
+        }
+
+        foreach (NeuronExtension *neuronExtension, m_neuronExtensions) {
+            ThingId ParentId = m_neuronExtensions.key(neuronExtension);
+            foreach (QString circuit, neuronExtension->digitalOutputs()) {
+                ThingDescriptor ThingDescriptor(digitalOutputThingClassId, QString("Digital output %1").arg(circuit), QString("Neuron extension %1, Slave address %2").arg(neuronExtension->type()).arg(QString::number(neuronExtension->slaveAddress())), ParentId);
```

Comment:
uh... this works?

The variable name is the same as the variable type...

Reply:
was a search and replace error